src/libasr/pass/python_bind.cpp

```diff
@@ -0,0 +1,406 @@
+        args_PyFloat_FromDouble.push_back(al,
+            {f.base.base.loc, ASRUtils::EXPR(ASR::make_Cast_t(al, f.base.base.loc, exp,
+                ASR::cast_kindType::RealToReal, f8_type, nullptr))});
+        conv_result = ASRUtils::EXPR(ASRUtils::make_FunctionCall_t_util(al, f.base.base.loc, sym_PyFloat_FromDouble,
+                        nullptr, args_PyFloat_FromDouble.p, args_PyFloat_FromDouble.n, ptr_t, nullptr, nullptr));
+    } else if (type->type == ASR::ttypeType::Character) {
+        ASR::symbol_t *sym_PyUnicode_FromString = parent_scope.get_symbol("PyUnicode_FromString");
+        Vec<ASR::call_arg_t> args_PyUnicode_FromString;
+        args_PyUnicode_FromString.reserve(al, 1);
+        args_PyUnicode_FromString.push_back(al, {f.base.base.loc, exp});
+        conv_result = ASRUtils::EXPR(ASRUtils::make_FunctionCall_t_util(al, f.base.base.loc,
+                        sym_PyUnicode_FromString, nullptr, args_PyUnicode_FromString.p, args_PyUnicode_FromString.n,
+                        ptr_t, nullptr, nullptr));
+    } else {
+        throw LCompilersException(
+            "Calling CPython with " + ASRUtils::get_type_code(ASRUtils::expr_type(exp)) + " type not supported");
+    }
+
+    LCOMPILERS_ASSERT(conv_result);
+    return conv_result;
+}
+
+void generate_body(Allocator &al, ASR::Function_t &f, SymbolTable &parent_scope) {
+    Vec<ASR::stmt_t*> body;
+    body.reserve(al, 1);
+    Str s;
+    
+    /*
+        if (!Py_IsInitialized()) {
+            Py_Initialize();
+            int pS = PyRun_SimpleString("import sys; sys.path.append('.')");
+        }
+    */
+    ASR::ttype_t *i4_type = ASRUtils::TYPE(ASR::make_Integer_t(al, f.base.base.loc, 4));
+
+    ASR::symbol_t *sym_Py_IsInitialized = parent_scope.get_symbol("Py_IsInitialized");
+    ASR::symbol_t *sym_Py_Initialize = parent_scope.get_symbol("Py_Initialize");
+    LCOMPILERS_ASSERT(sym_Py_IsInitialized)
+    ASR::asr_t *call_Py_IsInitialized = ASRUtils::make_FunctionCall_t_util(al, f.base.base.loc,
+                                            sym_Py_IsInitialized, nullptr, nullptr, 0, i4_type, nullptr, nullptr);
+    ASR::asr_t * if_cond = ASR::make_IntegerCompare_t(al, f.base.base.loc, ASRUtils::EXPR(call_Py_IsInitialized),
+                                ASR::cmpopType::Eq, ASRUtils::EXPR(ASR::make_IntegerConstant_t(al, 
+                                                        f.base.base.loc, 0, i4_type)), i4_type, nullptr);
+    Vec<ASR::stmt_t*> if_body;
+    if_body.reserve(al, 2);
+    ASR::asr_t *call_Py_Initialize = ASRUtils::make_SubroutineCall_t_util(al, f.base.base.loc, sym_Py_Initialize,
+                                        nullptr, nullptr, 0, nullptr, nullptr, false, false);
+    if_body.push_back(al, ASRUtils::STMT(call_Py_Initialize));
+
+    ASR::symbol_t *sym_PyRun_SimpleString = parent_scope.get_symbol("PyRun_SimpleString");
+    Vec<ASR::call_arg_t> args_PyRun_SimpleString;
+    s.from_str(al, "import sys; sys.path.append(\".\")");
```

Comment:
Yes, the `bindpy_05.py` example does not require this. But it is required for other tests example `bindpy_01.py`. 

We append the current path to CPython's module search path. If we do not do this, CPython will fail with `ModuleNotFoundError`, because it only looks for modules installed using pip or the builtins/CPython standard library.

Reply:
I think https://github.com/lcompilers/lpython/blob/9374feb17362d8046db1e4060d633f7ff9887100/integration_tests/bindpy_05.py#L72-L73 should help. I believe it adds the current directory `.` to the path.

Reply:
I can try this.

What does `pointer(argv1, i64)` do? What does the `pointer` function do?

Reply:
I figured it out.

I am no longer using `PyRun_SimpleString` and appending the path. I am using `Py_DecodeLocale` and `PySys_SetArgv`.